ir/src/tests/evaluators.rs

```diff
@@ -1,10 +1,10 @@
 use super::{parse, AirIR};
 
-#[ignore]
 #[test]
 fn simple_evaluator() {
     let source = "
     ev advance_clock(main: [clk]):
+        let z = a + 1
```

Comment:
Question: Shouldn't this be invalid but the test passes. I'm assuming that's what you were testing here. I think we currently don't validate the declaration unless we use the variable and this should be a bug. Did I miss something here or should we open an issue for this?

Reply:
Good point! I didn't even notice that actually. I think I just made a mistake. You're right though, we don't validate the declaration unless the variable is used. If you can open an issue, that would be great!

Reply:
Although really, this is just dead code, so validating it is not a priority. @bitwalker what's your take on these situations?

Reply:
@grjte I'm very much in favor of performing semantic analysis on all code, regardless of whether it is dead or not, but we need richer diagnostics infra for that to be worthwhile IMO. Otherwise I suspect it is easy to accidentally assume that something is working a certain way (because compilation succeeds) when it actually isn't. When we can do the diagnostics refactoring we discussed briefly, that would be a good time to introduce some of these additional warnings/errors.